Content.Client/UserInterface/Systems/Chat/ChatUIController.cs

```diff
@@ -845,6 +870,45 @@ public void Repopulate()
         }
     }
 
+    /// <summary>
+    /// Returns the chat name color for a mob
+    /// </summary>
+    /// <param name="name">Name of the mob</param>
+    /// <returns>Hex value of the color</returns>
+    public string GetNameColor(string name)
+    {
+        var colorIdx = Math.Abs(name.GetHashCode() % _chatNameColors.Length);
+        return _chatNameColors[colorIdx];
+    }
+
+    public string GetStringInsideTag(ChatMessage message, string tag)
+    {
+        var rawmsg = message.WrappedMessage;
+        var tagStart = rawmsg.IndexOf($"[{tag}]");
+        var tagEnd = rawmsg.IndexOf($"[/{tag}]");
+        if (tagStart < 0 || tagEnd < 0)
+            return "";
+        tagStart += tag.Length + 2;
+        return rawmsg.Substring(tagStart, tagEnd - tagStart);
+    }
```

Comment:
This is a copy-paste of `SpeechBubble.cs`'s `ExtractSpeechSubstring()`; it'd be acceptable to move it somewhere common that both can access to keep the code duplication down

Reply:
Done - they're now static methods in SharedChatSystem, matching the behavior of SanitizeAnnouncement.